### PR TITLE
Fix secret issues on CAPI

### PR DESCRIPTION
**Pull request by QuentinBisson**
## Checklist

I have:

- [x] Described why this change is being introduced
- [ ] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: service/controller/resource/certificates/resource.go [hunk 1]
```diff
@@ -21,11 +20,13 @@ import (
 type Config struct {
 	Name      string
 	Sources   []CertificateSource
-	Target    NameFunc
+	Target    NameFuncWithoutParam
 	K8sClient k8sclient.Interface
 	Logger    micrologger.Logger
 }
 
+type NameFuncWithoutParam func() string
```
Comment (QuentinBisson) [inline service/controller/resource/certificates/resource.go:28]: Add this to be able to remove key.SecretAPICertificates as this basically calls key.Secret without the cluster value

File: service/controller/resource/alerting/alertmanager/test/case-1-awsconfig.golden [hunk 2]
```diff
@@ -21,7 +21,7 @@ spec:
       app.kubernetes.io/instance: alertmanager
       app.kubernetes.io/managed-by: prometheus-meta-operator
       app.kubernetes.io/name: alertmanager
-  configSecret: new-alertmanager-config
+  configSecret: alertmanager-config
```
Comment (QuentinBisson) [inline service/controller/resource/alerting/alertmanager/test/case-1-awsconfig.golden:24]: This change is long overdue

File: service/controller/resource/resource.go [hunk 3]
```diff
@@ -97,10 +97,10 @@ func New(config Config) ([]resource.Interface, error) {
 				},
 				{
 					NameFunc:      key.CAPICertificateName,
-					NamespaceFunc: key.CAPICertificateNamespace,
```
Comment (QuentinBisson) [inline service/controller/resource/resource.go:100]: key.CAPICertificateNamespace == key.OrganizationNamespace

File: service/key/key.go [hunk 4]
```diff
@@ -217,7 +204,7 @@ func IsInCluster(installation string, obj interface{}) bool {
 	switch v := obj.(type) {
 	case *v1.Service:
 		return true
-	case *capiv1alpha3.Cluster:
```
Comment (QuentinBisson) [inline service/key/key.go:220]: This is the actual fix

**Comment by QuentinBisson** [comment]
Before the fix, mc prometheus were actually seen as workload clusters so the value the management cluster controller in pmo was copying the incorrect source secret.


Prometheus external labels on gubble after the fix:

```
externalLabels:
      cluster_id: gubble
      cluster_type: management_cluster
      customer: giantswarm
      installation: gubble
      pipeline: testing
      provider: openstack
      region: onprem
```

Prometheus external labels on halley before the fix:

```
externalLabels:
      cluster_id: halley
      cluster_type: workload_cluster
      customer: giantswarm
      installation: halley
      pipeline: stable
      provider: openstack
      region: onprem


```

**Review by mcbenjemaa — APPROVED** [review]

**Pull request merged**